prov/util/src/util_coll.c

```diff
@@ -733,7 +735,8 @@ static int util_coll_close(struct fid *fid)
 
 	coll_mc = container_of(fid, struct util_coll_mc, mc_fid.fid);
 
-	ofi_atomic_dec32(&coll_mc->av_set->ref);
+	if (ofi_atomic_dec32(&coll_mc->av_set->ref) == 0)
+		util_av_set_close(&coll_mc->av_set->av_set_fid.fid);
```

Comment:
This doesn't look correct either.  The caller should close the av set.  The ref cnt should be > 0 here.

Reply:
Remvoved this change as the new code is specific to av_close as it is limited to av->coll_mc.